extensions/amp-story/1.0/amp-story.js

```diff
@@ -713,6 +738,36 @@ export class AmpStory extends AMP.BaseElement {
       true /** callToInitialize */
     );
 
+    this.storeService_.subscribe(
+      StateProperty.SUBSCRIPTIONS_STATE,
+      (subscriptionsState) => {
+        this.subscriptionsStatePromise_.resolve();
+        if (subscriptionsState === SubscriptionsState.GRANTED) {
+          this.storeService_.dispatch(
+            Action.TOGGLE_SUBSCRIPTIONS_DIALOG_UI_STATE,
+            false
+          );
+        }
+      }
+    );
+
+    this.storeService_.subscribe(
+      StateProperty.SUBSCRIPTIONS_DIALOG_UI_STATE,
+      (dialogState) => {
+        if (!dialogState) {
+          this.paywallTimeout_ && clearTimeout(this.paywallTimeout_);
```

Comment:
need to set paywall timeout to null?

Reply:
nit: lets make this its own method. i.e. 
```js
if (!dialogState) {
  this.onSubcriptionDialogRemoval_() 
}
```

Reply:
Done.